packages/consumption/test/modules/attributes/AttributesController.test.ts

```diff
@@ -711,6 +740,304 @@ describe("AttributesController", function () {
+            expect(updatedPredecessorSharedCopy!.shareInfo!.sourceAttribute).toBeUndefined();
+        });
+
+        test("should delete predecessors of deleted attribute", async function () {
+            const attributeParams: ICreateRepositoryAttributeParams = {
+                content: IdentityAttribute.from({
+                    value: EMailAddress.from({
+                        value: "my@email.address"
+                    }),
+                    owner: consumptionController.accountController.identity.address
+                })
+            };
+            const predecessorRepositoryAttribute = await consumptionController.attributes.createRepositoryAttribute(attributeParams);
+
+            const repositorySuccessorParams: IAttributeSuccessorParams = {
+                content: IdentityAttribute.from({
+                    value: {
+                        "@type": "EMailAddress",
+                        value: "my-new@email.address"
+                    },
+                    owner: consumptionController.accountController.identity.address
+                })
+            };
+            const { successor: successorRepositoryAttribute } = await consumptionController.attributes.succeedRepositoryAttribute(
+                predecessorRepositoryAttribute.id,
+                repositorySuccessorParams
+            );
+
+            await consumptionController.attributes.executeFullAttributeDeletionProcess(successorRepositoryAttribute);
+            const result = await consumptionController.attributes.getLocalAttribute(predecessorRepositoryAttribute.id);
+            expect(result).toBeUndefined();
+        });
+
+        test("should change default from deleted attribute if another candidate exists", async function () {
+            const attributeParams: ICreateRepositoryAttributeParams = {
+                content: IdentityAttribute.from({
+                    value: EMailAddress.from({
+                        value: "my@email.address"
+                    }),
+                    owner: consumptionController.accountController.identity.address
+                })
+            };
+            const attributeParams2: ICreateRepositoryAttributeParams = {
+                content: IdentityAttribute.from({
+                    value: EMailAddress.from({
+                        value: "my2@email.address"
+                    }),
+                    owner: consumptionController.accountController.identity.address
+                })
+            };
+            const defaultRepositoryAttribute = await consumptionController.attributes.createRepositoryAttribute(attributeParams);
+            const otherRepositoryAttribute = await consumptionController.attributes.createRepositoryAttribute(attributeParams2);
+
+            await consumptionController.attributes.executeFullAttributeDeletionProcess(defaultRepositoryAttribute);
```

Comment:
Add a check that `defaultRepositoryAttribute` has `default: true` and that `otherRepositoryAttribute` has `default: undefined` before you delete `defaultRepositoryAttribute`.